superset/models/helpers.py

```diff
@@ -750,13 +750,30 @@ def columns(self) -> List[Any]:
     def get_fetch_values_predicate(self) -> List[Any]:
         raise NotImplementedError()
 
+    @property
+    def default_endpoint(self) -> sa.Text:
+        raise NotImplementedError()
+
+    @property
+    def type(self) -> str:
+        raise NotImplementedError()
+
+    @property
+    def id(self) -> int:
+        raise NotImplementedError()
+
     @staticmethod
     def get_extra_cache_keys(query_obj: Dict[str, Any]) -> List[str]:
         raise NotImplementedError()
 
     def get_template_processor(self, **kwargs: Any) -> BaseTemplateProcessor:
         raise NotImplementedError()
 
+    def explore_url(self) -> str:
```

Comment:
@hughhhh is this property needed in order to run a query?

Reply:
we may want to consider putting a constraint on slice.datasource_type so that we only save charts that have a datasource_type of table.

Reply:
> @hughhhh is this property needed in order to run a query?
it's not needed to run the for what I can see

> we may want to consider putting a constraint on slice.datasource_type so that we only save charts that have a datasource_type of table.
thats does seem proper to stop charts from being created with the wrong type

Reply:
Yeah, i think the issue here is that we are saving charts with a query datasource. I'm concerned that putting the value here for the explore mixin would muddy the waters between the functionality of exploring and the functionality of the base datasource, in this case the explore_url. I would recommend not adding this property to this class if it is not required for this functionality.